scripts/blocklist-skylink.sh

```diff
@@ -37,45 +34,18 @@ else
     skylinks=("$1") # just single skylink passed as input argument
 fi
 
-# Block skylinks in batches
-skylinks_len=${#skylinks[@]}
-for (( i = 0; i < $skylinks_len; i++ )); do
-    # Add skylink to batch
-    skylink="${skylinks[$i]}"
-    echo ".. ⌁ Adding skylink ${skylink} to batch..."
-    batch_skylinks+=("$skylink")
-
-    # For performance reasons on each iteration we do not block a single
-    # skylink, but we block skylinks in batches with BATCH_SIZE size mainly
-    # because of nginx cache search.
-    # If (batch len == batch size) or (we have last batch):
-    if (( ${#batch_skylinks[@]} == $BATCH_SIZE || $i == $skylinks_len - 1 )); then
-        echo "--------------------------------------------"
-
-        # Add to Sia blocklist
-        echo "Blocking batch skylinks in skyd..."
-        skylinks_space_separated="$(IFS=' '; echo "${batch_skylinks[*]}")"
-        docker exec sia siac skynet blocklist add $skylinks_space_separated
+# get local nginx ip adress
+nginx_ip=$(docker inspect -f '{{range.NetworkSettings.Networks}}{{.IPAddress}}{{end}}' nginx)
 
-        # Remove from NGINX cache
-        # NOTE:
-        # If there are changes to how the NGINX cache is being cleared, the same
-        # changes need to be applied to the /setup-scripts/blocklist-airtable.py
-        # script.
-        echo "Removing batch skylinks from Nginx cache..."
-        skylinks_pipe_separated="$(IFS='|'; echo "${batch_skylinks[*]}")"
-        cached_files_command="find /data/nginx/cache/ -type f | xargs -r grep -Els '^Skynet-Skylink: ($skylinks_pipe_separated)'"
-        docker exec -it nginx bash -c "${cached_files_command} | xargs -r rm"
+# iterate over provided skylinks and block them one by one
+for skylink in "${skylinks[@]}"; do
+    printf "Blocking ${skylink} ... "
+    status_code=$(curl --write-out '%{http_code}' --silent --output /dev/null --data "{\"add\":[\"$skylink\"]}" "http://${nginx_ip}:8000/skynet/blocklist")
 
-        # Clear batch
-        batch_skylinks=()
-
-        echo "--------------------------------------------"
+    # print blocklist response status code
+    if [ $status_code = "204" ]; then
```

Comment:
`nit` use `==`? it's a synonym right, but still 🤷‍♂️

Reply:
afaik `==` is `bash` specific alias and doesn't port to `ash` or some other shells so I think it's safer to use single quote just in case we want to run it on a different shell (?)